### PR TITLE
fix: explicitly handle arguments in create_model for sklearn and xgboost

### PR DESCRIPTION
**Pull request by aws-patlin**
*Issue #, if available:*
N/A

*Description of changes:*
Add explicit handling for `entry_point`, `source_dir`, and `dependencies` arguments in `create_model` for SKLearn and XGBoost to match other framework estimators. Improve handling for `image` argument as well.

*Testing done:*
Unit tests updated to check for handling these arguments. Unit tests run with `tox tests/unit`, all tests passing.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to any/all clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/sklearn/estimator.py [hunk 1]
```diff
@@ -185,17 +183,17 @@ def create_model(
...
             enable_cloudwatch_metrics=self.enable_cloudwatch_metrics,
             container_log_level=self.container_log_level,
             code_location=self.code_location,
             py_version=self.py_version,
             framework_version=self.framework_version,
             model_server_workers=model_server_workers,
-            image=image or self.image_name,
             sagemaker_session=self.sagemaker_session,
             vpc_config=self.get_vpc_config(vpc_config_override),
+            dependencies=(dependencies or self.dependencies),
```
Comment (ajaykarpur) [inline src/sagemaker/sklearn/estimator.py:196]: Should the unit test include `dependencies`?

File: src/sagemaker/xgboost/estimator.py [hunk 2]
```diff
@@ -147,28 +154,26 @@ def create_model(
...
+            source_dir=(source_dir or self._model_source_dir()),
             enable_cloudwatch_metrics=self.enable_cloudwatch_metrics,
             container_log_level=self.container_log_level,
             code_location=self.code_location,
             py_version=self.py_version,
             model_server_workers=model_server_workers,
-            image=self.image_name,
             sagemaker_session=self.sagemaker_session,
             vpc_config=self.get_vpc_config(vpc_config_override),
+            dependencies=(dependencies or self.dependencies),
```
Comment (ajaykarpur) [inline src/sagemaker/xgboost/estimator.py:176]: Same thing for XGBoost.

File: src/sagemaker/sklearn/estimator.py [hunk 3]
```diff
@@ -140,7 +140,14 @@ def __init__(
...
+        self,
+        model_server_workers=None,
+        role=None,
+        vpc_config_override=VPC_CONFIG_DEFAULT,
+        entry_point=None,
+        source_dir=None,
+        dependencies=None,
+        **kwargs
     ):
         """Create a SageMaker ``SKLearnModel`` object that can be deployed to an
```
Comment (ajaykarpur) [inline src/sagemaker/sklearn/estimator.py:152]: The docstring should be updated to include the new arguments.

File: src/sagemaker/xgboost/estimator.py [hunk 4]
```diff
@@ -125,7 +125,14 @@ def __init__(
...
+        self,
+        model_server_workers=None,
+        role=None,
+        vpc_config_override=VPC_CONFIG_DEFAULT,
+        entry_point=None,
+        source_dir=None,
+        dependencies=None,
+        **kwargs
     ):
         """Create a SageMaker ``XGBoostModel`` object that can be deployed to an ``Endpoint``.
```
Comment (ajaykarpur) [inline src/sagemaker/xgboost/estimator.py:137]: The docstring should be updated to include the new arguments.

**Comment by aws-patlin** [comment]
Looks like sagemaker-python-sdk-pr is failing due to ResourceLimitExceeded errors on a couple tests. @ajaykarpur can you restart it?

**Pull request merged**